tools/osv-linter/internal/checks/checks.go

```diff
@@ -0,0 +1,107 @@
+// Package checks defines and implements all checks and collections of checks.
+//
+// To add additional checks:
+// 1. define a new instance of `Check`
+// 2. add it to the `checks` array
+// 3. add it to the relevent collections defined in `checkCollections`
+//
+// To add additional collections of checks:
+// 1. add to the `checkCollections` array.
+package checks
+
+import (
+	"fmt"
+
+	"github.com/tidwall/gjson"
+)
+
+// CheckError describes when a check fails.
+type CheckError struct {
+	Code    string
+	Message string
+}
+
+// Error returns the error message, including the code.
+func (ce *CheckError) Error() string {
+	return fmt.Sprintf("%s: %s", ce.Code, ce.Message)
+}
+
+// CheckDef defines a single check.
+type CheckDef struct {
+	Code        string
+	Name        string
+	Description string
+	Check       Check
+}
+
+// Check defines how to run the check.
+type Check func(*gjson.Result) []CheckError
+
+// Run runs the check, returning any findings.
+// The check has no awareness of the check's Code,
+// this merges that with the check's findings.
+func (c *CheckDef) Run(json *gjson.Result) (findings []CheckError) {
+	for _, finding := range c.Check(json) {
+		findings = append(findings, CheckError{
+			Code:    c.Code,
+			Message: finding.Error(),
+		})
+	}
+	return findings
+}
+
+// CheckCollectionDef defines a named collection of checks.
+type CheckCollectionDef struct {
+	Name        string
+	Description string
+	Checks      []*CheckDef
+}
+
+// FromCode returns the check with a specific code.
+func FromCode(code string) *CheckDef {
+	for _, check := range CollectionFromName("ALL").Checks {
+		if check.Code == code {
+			return check
+		}
+	}
+	return nil
+}
+
+// FromName returns the check with a specific name.
+func FromName(name string) *CheckDef {
+	for _, check := range CollectionFromName("ALL").Checks {
+		if check.Name == name {
+			return check
+		}
+	}
+	return nil
+}
+
+var Collections = []CheckCollectionDef{
+	{
+		Name:        "ALL",
```

Comment:
Who is using ALL vs osv.dev here? Should we just have one collection for now to keep things simple?

Reply:
This collection definition is used by `FromName()` and `FromCode()`

Reply:
Got it, removed in 313d4c3